Core/src/org/sleuthkit/autopsy/modules/encryptiondetection/EncryptionDetectionDataSourceIngestModule.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.modules.encryptiondetection;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.casemodule.services.Blackboard;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProgress;
+import org.sleuthkit.autopsy.ingest.IngestModule;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModule;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestMessage;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.autopsy.ingest.ModuleDataEvent;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.Image;
+import org.sleuthkit.datamodel.ReadContentInputStream;
+import org.sleuthkit.datamodel.Volume;
+import org.sleuthkit.datamodel.VolumeSystem;
+
+/**
+ * Data source module to detect encryption.
+ */
+class EncryptionDetectionDataSourceIngestModule implements DataSourceIngestModule {
+
+    private final IngestServices services = IngestServices.getInstance();
+    private final Logger logger = services.getLogger(EncryptionDetectionModuleFactory.getModuleName());
+    private Blackboard blackboard;
+    private double calculatedEntropy;
+
+    private final double minimumEntropy;
+    
+    /**
+     * Create a EncryptionDetectionDataSourceIngestModule object that will detect
+     * volumes that are encrypted and create blackboard artifacts as appropriate.
+     * The supplied EncryptionDetectionIngestJobSettings object is used to
+     * configure the module.
+     */
+    EncryptionDetectionDataSourceIngestModule(EncryptionDetectionIngestJobSettings settings) {
+        minimumEntropy = settings.getMinimumEntropy();
+    }
+
+    @Override
+    public void startUp(IngestJobContext context) throws IngestModule.IngestModuleException {
+        try {
+            validateSettings();
```

Comment:
I am a fan of this not just to keep it consistent with the FileIngestModule, but also so that when we inevitably add more settings that need validating to the DataSourceIngestModule it will be more immediately clear where they should be validated.